travharv/execution_report.py

```diff
@@ -0,0 +1,199 @@
+        self.id = id
+        self.rdf_store_access = rdf_store_access
+        log.debug("AssertionReport initialized")
+        self.graph_reports = graph_reports
+
+    def report_to_store(self):
+        """
+        Report the results of the assertions to the store.
+        """
+        log.debug("Reporting the results of the assertionreport to the store")
+        log.debug(f"Subject URI: {self.subject_uri}")
+        log.debug(f"Assertion Path: {self.assertion_path}")
+        log.debug(f"Assertion Result: {self.assertion_result}")
+        log.debug(f"Assertion Time: {self.assertion_time}")
+        log.debug(f"ID: {self.id}")
+        log.debug(f"Graph Reports: {self.graph_reports}")
+        log.debug(f"Message: {self.message}")
+
+        for graph_report in self.graph_reports:
+            graph_report.report_to_store()
+
+        # TODO: write template for reporting triples
+        # to store with the report_content
+
+
+class TaskExecutionReport:
+    """
+    A class to represent a TaskExecutionReport.
+    This class will report the results of the assertions per task.
+    e.g.: All AssertionReports will be stored in a dictionary
+    """
+
+    def __init__(
+        self,
+        rdf_store_access: RDFStoreAccess,
+    ):
+        """constructor
+
+        :param task: Task
+        """
+        log.debug("TaskExecutionReport initialized")
+        self.report_content = {
+            "task_id": uuid4(),
+            "last_mod": datetime.now(),
+        }
+        self.assertion_reports = []
+        self.rdf_store_access = rdf_store_access
+
+    def report_to_store(self, assertion_report: AssertionReport):
+        """
+        Report the results of the assertions to the store.
+        """
+        log.debug(
+            "Reporting the results of the task execution report to the store"
+        )
+        assertion_report.report_to_store()
+        assertion_report_id = assertion_report.id
+        last_mod = assertion_report.assertion_time
+
+        log.debug(f"Assertion Report ID: {assertion_report_id}")
+        log.debug(f"Last Modified: {last_mod}")
+
+        self.assertion_reports.append(assertion_report)
+        # TODO: write template for adding assertion report to the task report
+
+
+class ExecutionReport:
+    """
+    A class to represent an ExecutionReport.
+    This class will report the results of the assertions per task per config.
+    e.g.: All TaskExecutionReports will be stored in a dictionary
+    """
+
+    def __init__(self, rdf_store_access: RDFStoreAccess, config_name: str):
+        """constructor
+
+        :param rdf_store_access: RDFStoreAccess
+        """
+        self.rdf_store_access = rdf_store_access
+        log.debug("ExecutionReport initialized")
+        self.report_content = {
+            "last_mod": datetime.now(),
+            "config_name": config_name,
+        }
+        self.task_reports = []
+
+    def report_to_store(self, task_execution_report: TaskExecutionReport):
```

Comment:
just don´t get this recurring pattern in these classes...

I don't get what the method is actively trying to do 

and at the least this suggests the name is not descriptive of what it thinks it doing?